### PR TITLE
Use prebuildify

### PR DESCRIPTION
**Pull request by ralphtheninja**
Closes https://github.com/Level/leveldown/issues/482 , closes https://github.com/Level/leveldown/issues/536

TODO

- [x] Add secure key for Travis
- [x] Add secure key for AppVeyor

@vweevers I'm not 100% sure about the changes to `appveyor.yml`. I've taken the setup from `prebuildify-ci` and `sodium-native` (which now uses travis for windows)

* https://github.com/mafintosh/prebuildify-ci/blob/master/index.js#L145-L187
* https://github.com/sodium-friends/sodium-native/blob/e203fd5306583ba4d719ca679c8f97333615ffbb/appveyor.yml

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -1,22 +1,48 @@
...
+build: false
+skip_branch_with_pr: true
 
 environment:
   matrix:
     - nodejs_version: "6"
     - nodejs_version: "8"
     - nodejs_version: "10"
+    - nodejs_version: "11"
+    - nodejs_version: "Current"
```
Comment (vweevers) [inline appveyor.yml]: This will slow the build down somewhat, as 11 === current

File: appveyor.yml [hunk 2]
```diff
@@ -1,22 +1,48 @@
...
 
+configuration: Release
 platform:
   - x86
   - x64
 
 install:
+  # Workaround for https://github.com/appveyor/ci/issues/2420
+  - SET PATH=%PATH%;C:\Program Files\Git\mingw64\libexec\git-core
```
Comment (vweevers) [inline appveyor.yml]: I don't think we need this (the snappy submodule has been working fine in builds so far)

File: .travis.yml [hunk 3]
```diff
@@ -2,9 +2,6 @@ sudo: false
 
 language: node_js
 
-before_install:
-  - export JOBS=max
```
[the remark this answers is not in the document]
Comment (ralphtheninja) [reply .travis.yml]: It's for `node-gyp` to build using all cores, but it doesn't work anymore.

File: package.json [hunk 4]
```diff
@@ -40,15 +39,14 @@
...
+    "uuid": "^3.2.1"
   },
   "scripts": {
     "install": "node-gyp-build",
-    "test": "standard && verify-travis-appveyor && nyc tape test/*-test.js && prebuild-ci",
+    "test": "standard && nyc tape test/*-test.js",
     "coverage": "nyc report --reporter=text-lcov | coveralls",
-    "rebuild": "prebuild --compile",
-    "prebuild": "prebuild --all --strip --verbose",
+    "rebuild": "node-gyp rebuild --verbose",
```
Comment (vweevers) [inline package.json]: `--verbose` might be too noisy when building locally? I almost never need it

File: .travis.yml [hunk 5]
```diff
@@ -13,5 +10,25 @@ node_js:
...
+  - tar --create --verbose --file="$ARCHIVE_NAME" --directory "$TRAVIS_BUILD_DIR/prebuilds" .
 
-after_success: npm run coverage
\ No newline at end of file
+deploy:
+  provider: releases
+  draft: false
+  prerelease: true
+  api_key:
+    secure: "FIXME!"
```
[the remark this answers is not in the document]
Comment (ralphtheninja) [reply .travis.yml]: Agreed!

File: appveyor.yml (hunk 1 above)
Comment (ralphtheninja) [reply appveyor.yml]: Lets remove 11 then?

File: .travis.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (ralphtheninja) [reply .travis.yml]: @vweevers We should be able to do the same for appveyor?

[the remark this answers is not in the document]
Comment (vweevers) [reply .travis.yml]: Yes.

[the remark this answers is not in the document]
Comment (ralphtheninja) [reply .travis.yml]: @vweevers I'm not sure about the syntax for env vars in appveyor.yml, mind having a look?

File: .travis.yml [hunk 6]
```diff
@@ -13,5 +10,25 @@ node_js:
...
+  - ARCHIVE_NAME="${TRAVIS_TAG:-latest}-$TRAVIS_OS_NAME-`uname -m`.tar"
+  - npm run prebuild
+  - tar --create --verbose --file="$ARCHIVE_NAME" --directory "$TRAVIS_BUILD_DIR/prebuilds" .
 
-after_success: npm run coverage
\ No newline at end of file
+deploy:
+  provider: releases
+  draft: false
+  prerelease: true
```
[the remark this answers is not in the document]
Comment (ralphtheninja) [reply .travis.yml]: I'm unsure of this setting. Just copied it. https://docs.travis-ci.com/user/deployment/releases/#advanced-options Doesn't say much imo.

File: .travis.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (ralphtheninja) [reply .travis.yml]: Aha, seems to mark releases as pre-releases. I'm not sure why we would want to do that.

[the remark this answers is not in the document]
Comment (ralphtheninja) [reply .travis.yml]: If you look at the releases for `sodium-native`, they are _all_ marked as pre-release https://github.com/sodium-friends/sodium-native/releases

[the remark this answers is not in the document]
Comment (ralphtheninja) [reply .travis.yml]: @vweevers Lets remove it?

[the remark this answers is not in the document]
Comment (ralphtheninja) [reply .travis.yml]: Gah, just realized I wrote my comments on the wrong "thread", sorry.

File: .travis.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (ralphtheninja) [reply .travis.yml]: See below.

[the remark this answers is not in the document]
Comment (vweevers) [reply .travis.yml]: > Lets remove it?

Yup

File: appveyor.yml [hunk 7]
```diff
@@ -1,22 +1,45 @@
...
+artifacts:
+  - path: prebuilds
+    name: $(APPVEYOR_REPO_TAG_NAME)-win-$(PLATFORM)
+    type: zip
+
+deploy:
+  - provider: GitHub
+    artifact: /.*\.zip/
+    auth_token:
+      secure: FIXME
```
Comment (vweevers) [inline appveyor.yml]: Not sure we can use a env var here; it's not documented. So let's just use https://ci.appveyor.com/tools/encrypt

File: .travis.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (vweevers) [reply .travis.yml]: > I'm not sure about the syntax for env vars in appveyor.yml, mind having a look?

Moved to a new thread :)

File: appveyor.yml (hunk 7 above)
Comment (ralphtheninja) [reply appveyor.yml]: @vweevers Can you take care of this?

**Pull request merged**